### PR TITLE
components in React

### PR DESCRIPTION
**Pull request by NosatskyiK**
Acceptance Criteria for 1

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  The ProductsList component correctly receives product data as props.
- [x]  ProductCard component created and renders a single product.
- [x]  Header and Footer elements are present.
- [x]  ProductList displays right on mobile devices

Acceptance Criteria for 1*
 

- [x] The Header component has clickable elements that toggle the display between the About and ProductsList components.
- [ ]  State management is handled in App.tsx, with state determining which component to show.
- [ ]  LocalStorage added to save items in card after page realod

https://nosatskyik.github.io/fe-react-2024/

### CODE REVIEW

File: src/components/cart/Cart.component.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { useState } from 'react';
+
+import cart from '@/assets/img/cart.svg';
+
+import styles from './cart.module.css';
+
+export const Cart = () => {
+    const [hasNotAdded, setAdded] = useState(false);
+
+    const toggleCart = () => {
```
Comment (vadym-zinchenko-moc) [inline src/components/cart/Cart.component.tsx]: it's extra method, you can use method `setAdded`

File: src/components/cart/Cart.component.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React, { useState } from 'react';
```
Comment (vadym-zinchenko-moc) [inline src/components/cart/Cart.component.tsx]: why you import `React`?

File: src/App.tsx [hunk 3]
```diff
@@ -1,14 +1,22 @@
+import React, { useState } from 'react';
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx]: why you import `React`?

File: src/components/cart/Cart.component.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import cart from '@/assets/img/cart.svg';
+
+import styles from './cart.module.css';
+
+export const Cart = () => {
+    const [hasNotAdded, setAdded] = useState(false);
```
Comment (vadym-zinchenko-moc) [inline src/components/cart/Cart.component.tsx]: Bad naming. You must have the same name method and variable just to add `set` 
example - `[count, setCount]`

File: src/components/productList/ProductList.component.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+import { ProductCard } from '@/components/productCard/ProductCard.component.tsx';
+import { productsData } from '@/data/ProductsData.ts';
+
+import styles from './productList.module.css';
+
+export const ProductList = () => (
+    <main className={styles.main}>
+        <ProductCard data={productsData} />
```
Comment (vadym-zinchenko-moc) [inline src/components/productList/ProductList.component.tsx]: `The ProductsList component correctly receives product data as props`

it seems that your `ProductsList`  must receive  productsData like props.
and `ProductCard` must receive product.

you can name `productsData` just like `products`

better to name props like original name of props  - `products={products}`

File: src/components/productList/ProductList.component.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+import { ProductCard } from '@/components/productCard/ProductCard.component.tsx';
+import { productsData } from '@/data/ProductsData.ts';
+
+import styles from './productList.module.css';
+
+export const ProductList = () => (
+    <main className={styles.main}>
```
Comment (vadym-zinchenko-moc) [inline src/components/productList/ProductList.component.tsx]: why do you use `<main>` tag here?

File: src/components/productList/productList.module.css [hunk 7]
```diff
@@ -0,0 +1,15 @@
+.main {
```
Comment (vadym-zinchenko-moc) [inline src/components/productList/productList.module.css]: why do you use `main` class here?

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

**Comment by NosatskyiK** [comment]
- Removed the unnecessary toggleCart function and replaced it with an arrow function in onClick.
- Removed unnecessary React imports
- Changed the name for useState in the cart
- Changed the productList logic
- Changed the main tag to a section and gave it the productList class

File: src/components/productCard/ProductCard.component.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+import { Cart } from '@/components/cart/Cart.component.tsx';
+import type { Product } from '@/interface/interfaceProductCategory.ts';
+
+import styles from './productCard.module.css';
+
+export const ProductCard: React.FC<{ product: Product }> = ({ product }) => (
+    <>
```
Comment (vadym-zinchenko-moc) [inline src/components/productCard/ProductCard.component.tsx]: extra Fragment

**Review by vadym-zinchenko-moc — APPROVED** [review]

**Review by vadym-zinchenko-moc — APPROVED** [review]

**Pull request merged**